rosidl_typesupport_opendds_c/resource/srv__type_support_c.cpp.em

```diff
@@ -132,71 +180,187 @@ static void * create_replier__@(service.namespaced_type.name)(
     DDS::Subscriber_var dds_subscriber,
     allocator_t allocator)
 {
-@# TODO: Implement, considering original code in ffe10f9 or earlier
-  return NULL;
+  return @('::'.join(service.namespaced_type.namespaces))::typesupport_opendds_cpp::create_replier__@(service.namespaced_type.name)(
+    dds_participant,
+    request_topic_str,
+    response_topic_str,
+    dds_publisher,
+    dds_subscriber,
+    allocator);
 }
 
 static const char * destroy_replier__@(service.namespaced_type.name)(
   void * untyped_replier,
   deallocator_t deallocator)
 {
-@# TODO: Implement, considering original code in ffe10f9 or earlier
-  return NULL;
+  return @('::'.join(service.namespaced_type.namespaces))::typesupport_opendds_cpp::destroy_replier__@(service.namespaced_type.name)(
+    untyped_replier, deallocator);
 }
 
 static bool take_request__@(service.namespaced_type.name)(
   void * untyped_replier,
   rmw_request_id_t * request_header,
   void * untyped_ros_request)
 {
-@# TODO: Implement, considering original code in ffe10f9 or earlier
-  return true;
+  using ReplierType = rosidl_typesupport_opendds_cpp::Replier<
+    @(__dds_msg_typesupport_type)RequestWrapper,
+    @(__dds_msg_typesupport_type)ResponseWrapper
+  >;
+  if (!untyped_replier || !request_header || !untyped_ros_request) {
+    return false;
+  }
+  
+  ReplierType * replier = static_cast<ReplierType *>(untyped_replier);
+
+  @(__dds_msg_typesupport_type)RequestWrapper dds_request_wrapper;
+
+  if (DDS::RETCODE_OK != replier->take_request(dds_request_wrapper)) {
+    fprintf(stderr, "take_request failed\n");
+    return false;
+  }
+
+  int64_t sequence_number =
+    (((int64_t)dds_request_wrapper.header().request_id().sequence_number().high) << 32) |
+    dds_request_wrapper.header().request_id().sequence_number().low;
+  request_header->sequence_number = sequence_number;
+
+  size_t IDENTITY_SIZE = 16;
+  OpenDDS::DCPS::GUID_t id = dds_request_wrapper.header().request_id().writer_guid();
+  std::memcpy(&(request_header->writer_guid[0]), &id, IDENTITY_SIZE);
```

Comment:
Why 16 for the `IDENTITY_SIZE`? Is there a better way to determine this size than a seemingly arbitrary literal?

Reply:
It's based on GUID size, used DDS wide. I don't see us changing it for RMW without change in the underlying DDS spec. Replaced with single definition in the header.

Reply:
GUIDs are always 16 bytes.  It's so fundamental that we don't even seem to have a constant for it in OpenDDS (aside from just using `sizeof`).  It would be good to see a `sizeof` expression, or a constant kept in a common header file.